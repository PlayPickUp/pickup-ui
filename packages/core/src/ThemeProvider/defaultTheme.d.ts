import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
