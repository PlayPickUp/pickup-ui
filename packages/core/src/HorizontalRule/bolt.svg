<svg width="15" height="18" viewBox="0 0 15 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M8.75469 0L8.28319 6.96H14.4L5.64531 18L6.11681 11.04H0L8.75469 0Z" fill="#615E66"/>
</svg>
