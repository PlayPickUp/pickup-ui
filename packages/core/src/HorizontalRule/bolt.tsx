import React from "react";

const Bolt = (props) => {
  return (
    <svg
      width="15"
      height="18"
      viewBox="0 0 15 18"
      fill={props.fill}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M8.75469 0L8.28319 6.96H14.4L5.64531 18L6.11681 11.04H0L8.75469 0Z"
        fill={props.fill}
      />
    </svg>
  );
};
export default Bolt;
