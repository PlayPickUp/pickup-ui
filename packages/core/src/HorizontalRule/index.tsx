import React from "react";
import { createUseStyles } from "react-jss";
import classNames from "classnames";

import Bolt from "./bolt";

import { DefaultTheme, HorizontalRuleProps } from "../types";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  root: {
    display: "flex",
    position: "relative",
    flexFlow: "row nowrap",
    justifyContent: "space-around",
    alignItems: "center",
    padding: theme.spacing.base * 4,
    "& * + *": {
      display: "inline-block",
    },
  },
  rule: {
    border: "none",
    backgroundColor: (props: HorizontalRuleProps) =>
      props.color ? props.color : theme.colors.grey.dark,
    height: 1,
    width: "100%",
  },
  bolt: {
    display: "inline-flex",
    justifyContent: "center",
    alignItems: "center",
    position: "relative",
    width: 120,
    textAlign: "center",
    "& img": {
      width: 14,
      height: "auto",
    },
  },
}));

const HorizontalRule: React.FC<HorizontalRuleProps> = ({
  showBolt,
  className,
  style,
  color,
}) => {
  const props: HorizontalRuleProps = {
    color: color,
  };
  const classes = useStyles({ ...props });
  return (
    <div className={classNames(classes.root, className)} style={style}>
      {showBolt ? (
        <>
          <hr className={classes.rule} />
          <div className={classes.bolt}>
            <Bolt fill={color} />
          </div>
          <hr className={classes.rule} />{" "}
        </>
      ) : (
        <hr className={classes.rule} />
      )}
    </div>
  );
};

export default HorizontalRule;
