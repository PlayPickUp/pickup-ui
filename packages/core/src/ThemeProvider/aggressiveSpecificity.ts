/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
