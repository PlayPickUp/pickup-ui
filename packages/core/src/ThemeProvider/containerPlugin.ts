/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
