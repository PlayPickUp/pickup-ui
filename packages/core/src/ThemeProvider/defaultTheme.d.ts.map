{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}