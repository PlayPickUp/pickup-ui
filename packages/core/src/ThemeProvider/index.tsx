import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
