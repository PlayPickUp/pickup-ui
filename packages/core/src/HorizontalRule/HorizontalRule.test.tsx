import React from "react";
import { render } from "@testing-library/react";

import ThemeProvider from "../ThemeProvider";
import HorizontalRule from ".";

test("Horizontal Rule renders without crashing, matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <HorizontalRule />
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("Horizontal Rule with Bolt renders without crashing, matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <HorizontalRule showBolt />
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("Color prop changes the color of the Horizontal Rule and bolt", () => {
  const { container } = render(
    <ThemeProvider>
      <HorizontalRule color="red" showBolt />
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("Props are passed and rendered", () => {
  const { getByTestId } = render(
    <ThemeProvider>
      <HorizontalRule className="test-1-2-3" style={{ opacity: 0.5 }} />
    </ThemeProvider>
  );
  // className
  expect(getByTestId("horizontalRule").getAttribute("class")).toContain(
    "test-1-2-3"
  );
  // style
  expect(getByTestId("horizontalRule").getAttribute("style")).toContain(
    "opacity: 0.5"
  );
});
