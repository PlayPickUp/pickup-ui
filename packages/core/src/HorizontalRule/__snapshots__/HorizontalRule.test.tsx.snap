// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color prop changes the color of the Horizontal Rule and bolt 1`] = `
<div>
  <div
    id="PickUpUI"
  >
    <div
      class="PU--root-0-2-1"
      data-testid="horizontalRule"
    >
      <hr
        class="PU--rule-0-2-2 PU--rule-d0-0-2-4"
      />
      <div
        class="PU--bolt-0-2-3"
      >
        <svg
          fill="red"
          height="18"
          viewBox="0 0 15 18"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8.75469 0L8.28319 6.96H14.4L5.64531 18L6.11681 11.04H0L8.75469 0Z"
            fill="red"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <hr
        class="PU--rule-0-2-2 PU--rule-d0-0-2-4"
      />
       
    </div>
  </div>
</div>
`;

exports[`Horizontal Rule renders without crashing, matches snapshot 1`] = `
<div>
  <div
    id="PickUpUI"
  >
    <div
      class="PU--root-0-2-1"
      data-testid="horizontalRule"
    >
      <hr
        class="PU--rule-0-2-2 PU--rule-d0-0-2-4"
      />
    </div>
  </div>
</div>
`;

exports[`Horizontal Rule with Bolt renders without crashing, matches snapshot 1`] = `
<div>
  <div
    id="PickUpUI"
  >
    <div
      class="PU--root-0-2-1"
      data-testid="horizontalRule"
    >
      <hr
        class="PU--rule-0-2-2 PU--rule-d0-0-2-4"
      />
      <div
        class="PU--bolt-0-2-3"
      >
        <svg
          height="18"
          viewBox="0 0 15 18"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8.75469 0L8.28319 6.96H14.4L5.64531 18L6.11681 11.04H0L8.75469 0Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <hr
        class="PU--rule-0-2-2 PU--rule-d0-0-2-4"
      />
       
    </div>
  </div>
</div>
`;
